WIP,DNM: tracing agg stats producer side

adityamaru opened the pull request.

adityamaru committed:
wire up tracing agg stats for event stream

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/109690)
<!-- Reviewable:end -->

adityamaru committed:
consumer side subscriber stats

adityamaru committed:
roachtest 50g

adityamaru committed:
add some ingestion stats

adityamaru committed:
add hist to ingestion performance stats

adityamaru committed:
server: differentiate between local and remote batch requests

adityamaru committed:
some more stats

adityamaru committed:
wip

adityamaru committed:
add flush time on destination

adityamaru committed:
wip

adityamaru committed:
add histogram for sst data size distribution

stevendanna committed:
streamingccl: support multiple processors per machine

Each stream ingestion processor has a number of components that we may
want to scale over time:

- The number of SQL connections used to stream data
- The number of goroutines used for key rewriting
- The number of goroutines used for batching and AddSSTabling keys

This PR gives us a rather inelegant hammer of scaling the whole lot at
once by putting multiple ingestion procesosrs on a single node.

If the node has been assigned multiple sources, we first create
additional workers by putting different source nodes on different
processors.

If there are still workers left over, we then find the source with the
most spans and divide up its work over the remaining workers.

In ad-hoc testing, this doesn't have a substantial impact on our TPCC
test case, so I've left the worker count set to 1 by default.

Epic: CRDB-26968

Release note: None

adityamaru committed:
bulk,*: add ProcessorID to the AggregatorEvent

It is possible that a single node may run multiple distsql processors,
each sending back their own aggregator stats. This has not been the case
so far in backup, restore or c2c but we should be prepared for
future changes that may introduce this. To that effect we add
the ProcessorID to the AggregatorEvent, so that the events are
namespaced by nodeID,flowID,processorID and events from different
processors on the same node don't overwrite each other.

Release note: None

adityamaru committed:
*: cleanup lock + unlock for new linter

adityamaru committed:
comment out addsstable

stevendanna committed:
WIP: split destination based on source.

See in-code comments for more commentary.

adityamaru committed:
verbose tracing for 4s addssts

adityamaru committed:
more fixup

adityamaru committed:
print ranges

adityamaru committed:
disable tracing and enable scatter

adityamaru committed:
3 procs

adityamaru committed:
Revert "admission: add metric for bypassed IO admission work"

This reverts commit 6dba9cc3daf1132d8224dabbd0fb005bea01811d.

adityamaru committed:
Revert "admission: add l0 control metrics"

This reverts commit 05f0064263ab9d58cc9c675597a1865cc4b157bb.

adityamaru committed:
Revert "admission: incorporate L0 size into overload score"

This reverts commit 6cf5cb82c2cd0171561d3d18e3e1850abc9ddc84.
